soc/silabs/silabs_s2/soc.c

```diff
@@ -22,6 +23,18 @@
 #include <sl_power_manager.h>
 #endif
 
+#if defined(CONFIG_PRINTK) || defined(CONFIG_LOG)
+#define PR_EXC(...) LOG_ERR(__VA_ARGS__)
+#else
+#define PR_EXC(...)
+#endif /* CONFIG_PRINTK || CONFIG_LOG */
+
+#if (CONFIG_FAULT_DUMP == 2)
+#define PR_FAULT_INFO(...) PR_EXC(__VA_ARGS__)
+#else
+#define PR_FAULT_INFO(...)
```

Comment:
I suggest to avoid to define macro that do not consume the arguments. You can get a lot of trouble with code like `PR_FAULT_INFO("counter %d\n", i++)`. You can also get unexpected warnings about unused variables/parameters depending how you compile the code.

There are several ways to fix that. One of them is:
```
#define PR_FAULT_INFO(...) if(0) { \
  PR_EXC(__VA_ARGS__); \
}
```

Reply:
Obviously my example with a macro does not work... We may have to define a function:
```
static inline void PR_FAULT_INFO(...)
{
    if (0) {  
        PR_EXC(__VA_ARGS__);
    }                
}

Reply:
These macros were copied verbatim from `arch/arm/core/cortex_m/fault.c`, and the usage in `smu_fault` mimics the implementations of `secure_fault`, `mem_manage_fault` etc in that file, since the SMU interrupt is effectively the bus version of `secure_fault`. I would prefer if they stay identical to the originals, even if they can theoretically give warnings -- the current usage doesn't trigger any warnings.